connmanager/outgoing_connections.go

```diff
@@ -29,10 +29,11 @@ func (c *ConnectionManager) checkOutgoingConnections(connSet connectionSet) {
 			return
 		}
 
-		c.addressManager.Attempt(address.NetAddress())
-		err := c.initiateConnection(address.NetAddress().TCPAddress().String())
+		netAddress := address.NetAddress()
+		c.addressManager.Attempt(netAddress)
+		err := c.initiateConnection(netAddress.TCPAddress().String())
```

Comment:
Can you put `netAddress.TCPAddress().String()` inside a variable and log it as well?